cmd/generate.go

```diff
@@ -81,6 +82,40 @@ func genProcessCluster(cmd *cobra.Command, clusterName string, p *ants.Pool) {
 		}
 	}
 
+	// list of components in cluster
+	for c, _ := range clusterComponents {
+		var clComps []string
+		clComps = append(clComps, c)
+
+		// only doing at cluster level as doing it in component process level was pain
+		if clusters == "" {
+			if components != "" {
+				break
+			}
+		}
+		d, err := os.Open(clusterDir)
+		if err != nil {
+			log.Fatal().Err(err).Msg("")
+		}
+		defer d.Close()
+		comDir, err := d.Readdirnames(-1)
+		if err != nil {
+			log.Fatal().Err(err).Msg("")
+		}
+		for _, dirName := range comDir {
+			for _, e := range clComps {
+				if !strings.Contains(dirName, e) {
+					delcomp := filepath.Join(clusterDir, dirName)
+					err = os.RemoveAll(delcomp)
+					if err != nil {
+						log.Fatal().Err(err).Msg("")
+					}
+				}
+			}
+		}
+		d.Close()
```

Comment:
this is already deferred above.

Reply:
without this, rendering for multiple clusters failed with `too many open files`

Reply:
updated